lib/plsql/jdbc_connection.rb

```diff
@@ -348,7 +348,7 @@ def plsql_to_ruby_data_type(metadata)
         [Java::OracleSql::CLOB, nil]
       when "BLOB"
         [Java::OracleSql::BLOB, nil]
-      when "NUMBER"
+      when "NUMBER", "BINARY_FLOAT", "BINARY_DOUBLE"
```

Comment:
After doing some reading I think it would be best to have Float(64bit) for BINARY_DOUBLE(64bit) and 32bit version of Float - new type for BINARY_FLOAT(32bit).
That could be done with approach as described here.
http://stackoverflow.com/questions/8169938/ruby-single-precision-number
Doing that change would be a challenge for me though. I could use some support or need to buy some more time.
I understand that the goal would be to have mathematical operations outcomes aligned between Ruby and Oracle for those datatypes so that:
2/3 in ruby using Float = 2/3 in Oracle using BINARY_DOUBLE and
2/3 in ruby using single-Float = 2/3 in Oracle using BINARY_FLOAT.
Would that be a good expectation?